src/app/dialog/dialog.service.ts

```diff
@@ -40,34 +43,33 @@ import {Store} from '@ngrx/store';
 })
 export class DialogService {
   public callback: any;
-  private open: boolean;
 
-  public constructor(private router: Router, private store: Store<AppState>, private route: ActivatedRoute) {
+  private open$: Observable<boolean>;
+
+  public constructor(private router: Router, private store: Store<AppState>) {
     if (environment.videoKey) {
       this.store.dispatch(new VideosAction.LoadVideos({videos: getAllVideos(), apiKey: environment.videoKey}));
     }
+
+    this.open$ = this.bindOpen();
```

Comment:
Do we need this property still there?

Reply:
No, but we might forget about these problems in a few months. I added it here in order to allow us to access dialog state in the future without implementing nasty workarounds as we did in the past.

Reply:
Ok